docs/release-notes/release-notes-0.17.0.md

```diff
@@ -49,6 +49,16 @@ package](https://github.com/lightningnetwork/lnd/pull/7356)
   a helpful note-to-self containing arbitrary useful information about the
   channel.
 
+* [gRPC keepalive parameters can now be set in the
+  configuration](https://github.com/lightningnetwork/lnd/pull/7730). This also
+  lowers the default minimum time the server requires the client to wait between
+  sending pings. So clients are encouraged to set `grpc.keepalive_time_ms` when
+  creating long open streams over a network topology that might silently fail
+  connections. By default, clients now only need to wait 5 seconds between pings
+  (was 5 minutes before), so a value of `grpc.keepalive_time_ms=5100` is
```

Comment:
do you mean `grpc.client-ping-min-wait=5100ms`?

Reply:
No, `grpc.client-ping-min-wait` dictates how often the server allows a ping from the client. So if the client pings too often, the server will disconnect. But to tell the client that the server allows it to ping more often, the `grpc.keepalive_time_ms` value has to be lowered on the client side so it actually checks the connection more often. Not all clients set this, so I think for normal `lndclient`/gRPC connections no ping happens at all.

Reply:
was confused as it looked like a js specific setting to me, but I guess it's used everywhere.

> But to tell the client that the server allows it to ping more often, the grpc.keepalive_time_ms value has to be lowered on the client side so it actually checks the connection more often.

Not sure I follow. So in order for the **server** to tell the **client** that the server allows more frequent pings, the **client** needs to lower the `grpc.keepalive_time_ms`?

Reply:
The thing is, every gRPC library in every programming language has a different way to identify the client ping time.
For Golang it's:
```
			grpc.WithKeepaliveParams(keepalive.ClientParameters{
				Time: defaultClientPingTime,
			}),
```
(see https://github.com/lightninglabs/pool/pull/457).

But for JavaScript it's actually:
```
let lightning = new lnrpc.Lightning(
    'localhost:10019', credentials, {'grpc.keepalive_time_ms': 5500},
);
```
and for C++ it's yet another way. So not sure how to best communicate the setting's name. But it looks like `grpc.keepalive_time_ms` gives the most search results on Google so it seems to be pretty well understood.

> So in order for the server to tell the client that the server allows more frequent pings, the client needs to lower the grpc.keepalive_time_ms?

Not quite. By default, the client does not ping at all and the server only allows pings **from** the client every 5 minutes (which is a bit slow for our purposes). So with the change in this PR we configure the server to allow the client to ping way more often. BUT to actually initiate the pinging, the **client** also needs to enable the keepalive.

I'm not sure why, but it looks like in the setup I tested (`lnd` with a Node.js client), the server didn't actually ping the client (or it was not logged for some reason). Debugging this isn't easy, especially with an encrypted connection. But it looked like the client initiated pings were going through regularly, so that seems to be the most reliable option.

Reply:
The LND server pings ever 2 hours by default (at least before this PR), so you probably didn't wait long enough to see that happen?

Also, even if the LND server pings every 2 hours, the client doesn't know how frequent the server will be pinging, so the client will still not disconnect after 2 hours have gone by with no ping. Instead, the client will disconnect after the time has passed that it is configured to send pings to the server. However, I'm not sure if the server sends pings as quickly as the client plans to send them, will the client count the server's ping as it's own, or will they both be pinging?

Reply:
For python, the option is `grpc.keepalive_time_ms`.

Reply:
Cool thanks for the discussion guys!